### PR TITLE
OpenVR rendering & Nsight instrumentation improvements 

### PR DESCRIPTION
**Pull request by hyperlogic**
* updated to openvr 0.9.19 SDK, this seems to make the touch pad more reliable
* moved WaitGetPoses after main window preview.
* Added nsight support for color and payload in ranges
* Added ranges for present thread and application idle, physics, avatars & animation
* Rendering for non-threaded openvr present has rock solid tracking *if*, and this is a big if, our idle and render take under 8ms.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 209524a5c08441d9df6e10e232ca2b3f8c098548 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c40b70688cbee4345ae720df7844fa7fc5694391 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9dd24f81d4ea8a7f213c1b6786b92ed97470105c available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -2496,6 +2496,8 @@ static uint32_t _renderedFrameIndex { INVALID_FRAME };
 
 void Application::idle(uint64_t now) {
 
+    PROFILE_RANGE(__FUNCTION__);
```
Comment (jherico) [inline interface/src/Application.cpp]: I'm not sure about this.  idle gets called *way* more than it actually does anything, to the extent that I'd be concerned about the overhead of the nvidia functions.

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -3138,6 +3139,9 @@ void Application::updateDialogs(float deltaTime) {
 }
 
 void Application::update(float deltaTime) {
+
+    PROFILE_RANGE_EX(__FUNCTION__, 0xffff0000, (uint64_t)getActiveDisplayPlugin()->presentCount());
```
Comment (jherico) [inline interface/src/Application.cpp]: why are we using present count here?  The present count is asynchronous versus the main application... it seems like we'd want `_frameCount`

File: plugins/openvr/src/OpenVrDisplayPlugin.h [hunk 3]
```diff
@@ -35,6 +35,7 @@ class OpenVrDisplayPlugin : public HmdDisplayPlugin {
 
     void hmdPresent() override;
     bool isHmdMounted() const override;
+    void postPreview() override;
```
Comment (jherico) [inline plugins/openvr/src/OpenVrDisplayPlugin.h]: I don't see this function defined anywhere.

File: libraries/gl/src/gl/GLEscrow.h [hunk 4]
```diff
@@ -71,11 +71,16 @@ class GLEscrow {
         }
 
         bool signaled() const {
```
Comment (jherico) [inline libraries/gl/src/gl/GLEscrow.h]: this should be reverted...  if the client application wants to fetch the most recent item and trigger a GPU wait, it should call `fetchAndReleaseWithGpuWait`.  Additionally, `signaled` is used by the recycling code, which releases textures back to the main thread for re-use, so putting the sync here will cause unnecessary GPU waits on the rendering thread.

**Comment by jherico** [comment]
I'd suggest reverting the changes to the logic, but keeping the addition of the new profile range ctor and instances, and then opening the logic changes in a separate PR.

File: interface/src/Application.cpp (hunk 1 above)
Comment (hyperlogic) [inline interface/src/Application.cpp]: Yep, I agree having this here results in many calls.  I just wanted to see how often the idle was getting called, which was about every 25 microseconds.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8bb387dc740734b426573f4d77dda8615d20bcf9 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 73725106f0f6824d09ea3bb29704348fc54b21c2 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88c4e54cff936f7c4aa96bb1196f7163c1db2a12 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by hyperlogic** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88c4e54cff936f7c4aa96bb1196f7163c1db2a12 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: de3144cde50d510581084f776c51f8d0ee665ea8 available at: https://gist.github.com/dc8b3d99cd400dd5f08f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7431.dmg

**Comment by cozza13** [comment]
tested on the vive QA pass

**Pull request merged**